docker-plugin/src/test/java/com/nirima/jenkins/plugins/docker/builder/DockerBuilderPublisherTest.java

```diff
@@ -0,0 +1,100 @@
+package com.nirima.jenkins.plugins.docker.builder;
+
+import static org.hamcrest.CoreMatchers.containsString;
+
+import org.junit.Assert;
+import org.junit.Test;
+
+public class DockerBuilderPublisherTest {
+    private static final String VALID1 = "myimage";
+    private static final String VALID2 = "my/image-2";
+    private static final String VALID3 = "myimage_3:tag";
+    private static final String VALID4 = "myimage.4:Tag4";
+    private static final String VALID5 = "localhost/myimage";
+    private static final String VALID6 = "localhost.localdomain/myimage:My_Tag-6";
+    private static final String VALID7 = "my.repository.com:1234/myimage.7";
+    private static final String VALID8 = "my.repository.com:${PORT_NUMBER}/myimage.7";
+    private static final String VALID9 = "localhost.localdomain:5000/myimage:${JOB_NAME}-latest";
+    private static final String VALID10 = "localhost.localdomain:5000/myimage:${JOB_NAME}-${BUILD_NUMBER}";
+    private static final String INVALID1 = "MyImagei1";
+    private static final String INVALID2 = "myimage%";
+    private static final String INVALID3 = "1.2.3.4:abc/myimage:invalid3";
+    private static final String INVALID4 = "funnyhost�name:5000/myimage4";
```

Comment:
Apologies for the encoding issue - I was developing on a Windows PC using Eclipse, and although I've set things to UTF-8 in the past I guess something's defaulting to Windows encoding again.
As you probably guessed, the purpose of the INVALID4 name was to have an illegal hostname so I could verify that my regex was working the way I expected it to.